platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -39,23 +39,21 @@ root ::= statements
 statement ::= (assignment | expression)
 statements ::= (statement semis?)*
 
-assignment ::= directlyAssignableExpression EQ expression {pin=2}
+assignment ::= directlyAssignableExpression EQ primaryExpression {pin=2}
 expression ::= postfixUnaryExpression
 
 semis ::= SEMI+
 
 directlyAssignableExpression ::=
-    (postfixUnaryExpression navigationSuffix*) |          // navigationSuffix changed to optional due to greedy matching
+    primaryExpression |
     simpleIdentifier |
     parenthesizedDirectlyAssignableExpression
 
 parenthesizedDirectlyAssignableExpression ::= LP directlyAssignableExpression RP
 
-postfixUnaryExpression ::= primaryExpression postfixUnarySuffix*
+postfixUnaryExpression ::= primaryExpression postfixUnarySuffixes?
 
-postfixUnarySuffix ::= callSuffix | navigationSuffix
-
-callSuffix ::= (valueArguments? annotatedLambda) | valueArguments
+postfixUnarySuffixes ::= (valueArguments? annotatedLambda) | (navigationSuffix* valueArguments)
 
```

Comment:
Just a thought: what do you think about documenting the path of the changes applied to this grammar starting from the official Kotlin grammar – perhaps right in the comments in this file? For anyone comparing the grammars (perhaps in an attempt to implement a parser or reuse a Kotlin parser) it would be very helpful – at least the way that the existing rules have changed.

Reply:
Good idea, I'll try to make it happen.